src/main/java/org/swisspush/redisques/util/DefaultRedisProvider.java

```diff
@@ -58,23 +58,23 @@ private Future<RedisAPI> setupRedisClient(){
 
     private Future<RedisAPI> connectToRedis() {
         RedisquesConfiguration config = configurationProvider.configuration();
-        String redisHost = config.getRedisHost();
-        int redisPort = config.getRedisPort();
         String redisAuth = config.getRedisAuth();
-        boolean redisEnableTls = config.getRedisEnableTls();
+        String redisUser=config.getRedisUser();
         int redisMaxPoolSize = config.getMaxPoolSize();
         int redisMaxPoolWaitingSize = config.getMaxPoolWaitSize();
         int redisMaxPipelineWaitingSize = config.getMaxPipelineWaitSize();
 
         Promise<RedisAPI> promise = Promise.promise();
-        String protocol =  redisEnableTls ? "rediss://" : "redis://";
-        Redis.createClient(vertx, new RedisOptions()
-                .setConnectionString(protocol + redisHost + ":" + redisPort)
-                .setPassword((redisAuth == null ? "" : redisAuth))
+        RedisOptions redisOptions = new RedisOptions()
+                .setConnectionString(createConnectString(redisUser, redisAuth))
```

Comment:
What is the difference between setting the password in the connection string and through _.setPassword_ method? What is the redisUser? We only have the password which is defined in the _redisAuth_ property

Reply:
The password set by setPassword is the legacy authentication method. This is configure in redis.conf with requirepass. The connectString with username and password was introduced with Redis 6 (ACL). When running Redis on AWS we don't have the legacy mode available and therefore we must be capable to use the new way. Of course the legacy method still has to work. 
See https://redis.io/docs/management/security/ and https://redis.io/docs/management/security/acl/ for details.

Reply:
OK. For us, we still rely on the legacy mode to work, since we are not on Redis 6 yet.